File: src/main/java/cz/kamenitxan/jakon/utils/validation/PebbleExtension.java
package cz.kamenitxan.jakon.utils.validation;

import com.mitchellbosecke.pebble.extension.AbstractExtension;
import com.mitchellbosecke.pebble.extension.Function;
import cz.kamenitxan.jakon.utils.GetAttributeFun;
import cz.kamenitxan.jakon.utils.GetAttributeTypeFun;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by tomaspavel on 6.10.16.
 */
public class PebbleExtension extends AbstractExtension {

	@Override
	public Map<String, Function> getFunctions() {
		return new HashMap<String, Function>() {{
			put("getAttr", new GetAttributeFun());
			put("getAttrType", new GetAttributeTypeFun());
		}};
	}
}

File: modules/backend/src/main/scala/cz/kamenitxan/jakon/core/template/pebble/PebbleExtension.scala
package cz.kamenitxan.jakon.core.template.pebble

import cz.kamenitxan.jakon.webui.functions.LinkFun
import com.mitchellbosecke.pebble.extension.{AbstractExtension, Filter, Function}

import java.util

/**
  * Created by tomaspavel on 6.10.16.
  */
class PebbleExtension extends AbstractExtension {
	override def getFunctions: util.Map[String, Function] = {
		val extensions = new util.HashMap[String, Function]()
		extensions.put("i18n", new I18nFun)
		extensions.put("link", new LinkFun)
		extensions.put("value", new ValueFun)
		extensions
	}

	override def getFilters: util.Map[String, Filter] = {
		val extensions = new util.HashMap[String, Filter]()
		extensions.put("date", new JakonDateFilter)
		extensions.put("md", new MarkdownFilter)
		extensions.put("asJava", new AsJavaFilter)
		extensions
	}
}